### PR TITLE
OSDOCS-2082: Add coreos print-stream-json sub-command to AWS AMI install

### PR DESCRIPTION
**Pull request by bobfuru**
[OSDOCS-2082](https://issues.redhat.com/browse/OSDOCS-2082)
In this example, we want to give admins a better way than the current workflow of hardcoding AMIs that are copy-pasted from the openshift-installer. This PR documents how to run the new coreos sub-command of the installer so that users can avoid having to manually parse the HTML in `modules/installation-aws-user-infra-rhcos-ami.adoc`.

**Preview links:**
- [Printing RHCOS AMIs for AWS zones by using stream metadata](https://deploy-preview-33415--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-restricted-networks-aws.html#installation-aws-ami-stream-metadata_installing-restricted-networks-aws) (AWS UPI install in restricted network)
- [ Printing RHCOS AMIs for AWS zones by using stream metadata ](https://deploy-preview-33415--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-user-infra.html#installation-aws-ami-stream-metadata_installing-aws-user-infra)(AWS UPI install with CloudFormation templates)

### CODE REVIEW

**Comment by bobfuru** [comment]
Hi @cgwalters (Eng) and @miabbott (QE) - PTAL for review in OCP 4.8. 🙇 

This content for bootimages is generally new to me, so I've looked at the discussion in the stream-metadata-go library and in the [GRPA-1807](https://issues.redhat.com/browse/GRPA-1807?focusedCommentId=16310109&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16310109) epic, along with the [CoreOS bootimages doc](https://github.com/openshift/installer/commit/b64d699a13669e36e85e4cfe6fa6313202a7a690#diff-8749db54915c027b82a257627c6183c8edeb0c3b24a4b4a21a0bce8e53f180e9R88) in the `openshift/installer` repo.

This PR starts with the UPI install docs for automating the AWS AMI bootimage using the `openshift-install coreos print-stream-json` command. You'll see also where I've left the docs intact for manually specifying the AMI (aside from a few minor wording changes).

Once the AMI use case in this PR has been resolved, I can look at adding manual copy instructions for OpenStack QCOW2 and bare metal ISO in a new PR.

File: modules/installation-aws-ami-stream-metadata.adoc
Comment (bobfuru) [inline modules/installation-aws-ami-stream-metadata.adoc:10]: Alternate title: "Accessing RHCOS AMIs with stream metadata"
Suggestions for better wording welcome!

File: modules/installation-aws-ami-stream-metadata.adoc
Comment (bobfuru) [inline modules/installation-aws-ami-stream-metadata.adoc:16]: I'm not clear about what "the `openshift-install` binary contains pinned versions of bootimages" means (esp. "pinned"), so if this can or should be clarified, please do so.

File: modules/installation-aws-ami-stream-metadata.adoc
Comment (bobfuru) [inline modules/installation-aws-ami-stream-metadata.adoc:30]: Gathered from lines [99-100](https://github.com/openshift/installer/commit/b64d699a13669e36e85e4cfe6fa6313202a7a690#diff-8749db54915c027b82a257627c6183c8edeb0c3b24a4b4a21a0bce8e53f180e9). I couldn't test this step, so please verify that this is the expected command syntax and output. 

Also, are there any other explicit steps that one should take from here? Or should we say anything about how to use the AMI output in scripting for automation?

File: modules/installation-aws-user-infra-rhcos-ami.adoc
Comment (bobfuru) [inline modules/installation-aws-user-infra-rhcos-ami.adoc:9]: Debated whether to include a note before or after this table that using stream metadata is the recommended method for easier and more reliable automation processes?

File: modules/installation-aws-ami-stream-metadata.adoc
Comment (cgwalters) [inline modules/installation-aws-ami-stream-metadata.adoc:16]: I'd invert these and put this first, and then recommend that if you want to parse it from a Go program, use the library. 

Then, also mention that parsing it via `jq` or any other tooling that handles JSON (e.g. any other programming language's JSON library) is also fine.

File: modules/installation-aws-ami-stream-metadata.adoc
Comment (cgwalters) [reply modules/installation-aws-ami-stream-metadata.adoc]: How about: 
> The `openshift-install` binary contains references to the version of {op-system} bootimages that are tested for use with OpenShift, such as the AWS AMI.

File: modules/installation-aws-ami-stream-metadata.adoc
Comment (bobfuru) [reply modules/installation-aws-ami-stream-metadata.adoc]: Whoops, I might have been on 4.7 binary. It worked of course with 4.8.

I haven't been involved in the manual update to the AMI table before (e.g., https://github.com/openshift/openshift-docs/pull/29337) but this is really cool. I ran it and could get all 4.8 AMIs:
```
$ stream openshift-install coreos print-stream-json | jq -r '.architectures.x86_64.images.aws.regions[].image'           
ami-08c63bb1b8b661ab9
ami-01f2a0e8aee56e1f4
ami-012420354c0b2bf59
ami-0d8c0ce731da9faaf
ami-00cac27e65ae02107
ami-05b69d7e80f1178c6
ami-0bd576d6deb6094c8
ami-06e7e674d4b8cdc59
ami-04990181ab3872dc9
ami-0ed56edc094183f18
ami-0e63e7e6de6dac812
ami-01c63dbdcd28938c0
ami-04885ff040bf8e8d0
ami-04b31bbfe532cbb45
ami-0e188558a41f87d82
ami-006b4ea9d3e62fdc2
ami-094888ec0f3125ad7
ami-0b08ba24af2f005c9
ami-0cd5c6a0f8bb3c33d
ami-0d3e625f84626bbda
ami-040e8d23c125e7a75
```
Cc: @codyhoag @kalexand-rh

Comment (bobfuru) [reply modules/installation-aws-ami-stream-metadata.adoc]: Or just the `us-west-1` zone as described in the procedure:
`$ openshift-install coreos print-stream-json | jq -r '.architectures.x86_64.images.aws.regions["us-west-1"].image'`
`ami-0d3e625f84626bbda`

**Review by bobfuru — COMMENTED** [review]
Thanks for the review, @cgwalters. I made some revisions based on your feedback. Could you PTAL at these updates?

File: modules/installation-aws-ami-stream-metadata.adoc
Comment (miabbott) [inline modules/installation-aws-ami-stream-metadata.adoc:24]: nit:  `s/zone/region/`

File: modules/installation-aws-ami-stream-metadata.adoc
Comment (miabbott) [inline modules/installation-aws-ami-stream-metadata.adoc:22]: This statement is confusing.  `jq` is just a command line utility, not a JSON library.

Looking at Colin's feedback, I think you could say something like this:

```
You can also parse the stream metadata using the JSON library of your preferred programming language or via a command line utility that handles JSON data like `jq`.
```

File: modules/installation-aws-ami-stream-metadata.adoc
Comment (miabbott) [inline modules/installation-aws-ami-stream-metadata.adoc:37]: nit: `s/zone/region/`

File: modules/installation-aws-user-infra-rhcos-ami.adoc
Comment (miabbott) [reply modules/installation-aws-user-infra-rhcos-ami.adoc:9]: nit: `s/zones/regions/`

**Comment by miabbott** [comment]
Overall content looks good, just some suggested cleanups/corrections.

File: modules/installation-aws-ami-stream-metadata.adoc
Comment (bobfuru) [reply modules/installation-aws-ami-stream-metadata.adoc]: Thanks, I was struggling with the wording because as a procedure, I wanted to make it task-based ("Do this" as opposed to "you can do x or y"). Reworded a bit so that hopefully it makes more sense and is still is directive enough to be part of a procedure.

**Comment by bobfuru** [comment]
Applied feedback from @miabbott and will wait for approval from @cgwalters before merging. Thanks!

**Comment by ahardin-rh** [comment]
LGTM!

**Review by cgwalters — APPROVED** [review]
/lgtm

**Pull request merged**
